release/serve_tests/workloads/autoscaling_multi_deployment_1k.py

```diff
@@ -0,0 +1,203 @@
+                request, depth + 1)
+
+            return await obj_ref
+
+        async def __call__(self, request):
+            return await self.handle_request(request, 0)
+
+    for deployment in all_deployment_names:
+        Echo.options(name=deployment).deploy()
+
+
+@click.command()
+@click.option("--min-replicas", "-min", type=int)
+@click.option("--max-replicas", "-max", type=int)
+@click.option("--num-deployments", "-nd", type=int)
+@click.option("--trial-length", "-tl", type=str)
+def main(min_replicas: Optional[int], max_replicas: Optional[int],
+         num_deployments: Optional[int], trial_length: Optional[str]):
+    # Give default cluster parameter values based on smoke_test config
+    # if user provided values explicitly, use them instead.
+    # IS_SMOKE_TEST is set by args of releaser's e2e.py
+    smoke_test = os.environ.get("IS_SMOKE_TEST", "1")
+    if smoke_test == "1":
+        min_replicas = min_replicas or DEFAULT_SMOKE_TEST_MIN_NUM_REPLICA
+        max_replicas = max_replicas or DEFAULT_SMOKE_TEST_MAX_NUM_REPLICA
+        num_deployments = num_deployments or DEFAULT_SMOKE_TEST_NUM_DEPLOYMENTS
+        trial_length = trial_length or DEFAULT_SMOKE_TEST_TRIAL_LENGTH
+        logger.info(f"Running smoke test with min {min_replicas} and max "
+                    f"{max_replicas} replicas, {num_deployments} deployments "
+                    f".. \n")
+        # Choose cluster setup based on user config. Local test uses Cluster()
+        # to mock actors that requires # of nodes to be specified, but ray
+        # client doesn't need to
+        num_nodes = int(math.ceil(max_replicas / NUM_CPU_PER_NODE))
+        logger.info(
+            f"Setting up local ray cluster with {num_nodes} nodes .. \n")
+        serve_client = setup_local_single_node_cluster(num_nodes)[0]
+    else:
+        min_replicas = min_replicas or DEFAULT_FULL_TEST_MIN_NUM_REPLICA
+        max_replicas = max_replicas or DEFAULT_FULL_TEST_MAX_NUM_REPLICA
+        num_deployments = num_deployments or DEFAULT_FULL_TEST_NUM_DEPLOYMENTS
+        trial_length = trial_length or DEFAULT_FULL_TEST_TRIAL_LENGTH
+        logger.info(f"Running full test with min {min_replicas} and max "
+                    f"{max_replicas} replicas, {num_deployments} deployments "
+                    f".. \n")
+        logger.info("Setting up anyscale ray cluster .. \n")
+        serve_client = setup_anyscale_cluster()
+
+    http_host = str(serve_client._http_config.host)
+    http_port = str(serve_client._http_config.port)
+    logger.info(f"Ray serve http_host: {http_host}, http_port: {http_port}")
+
+    logger.info(f"Deploying with min {min_replicas} and max {max_replicas}"
+                f"target replicas ....\n")
+    setup_multi_deployment_replicas(min_replicas, max_replicas,
+                                    num_deployments)
+
+    logger.info("Warming up cluster ....\n")
+    rst_ray_refs = []
```

Comment:
What is `rst`?

Reply:
I'm not sure– I copied that over from `multi_deployment_1k_noop_replica.py`. It stores references to endpoints that we warm up, but I'm unsure what the `rst` stands for. I can refactor it.